src/discourse.py

```diff
@@ -0,0 +1,285 @@
+"""Gateway for accessing the Discourse API (for forums)"""
+
+import json
+import re
+from urllib import urlencode
+
+from google.appengine.api import urlfetch
+from google.appengine.ext import ndb
+
+
+class Error(Exception):
+    pass
+
+
+class DiscourseAPIClient(object):
+    """An API client for interacting with Discourse"""
+
+    def __init__(self, discourse_url, api_key, api_username='system'):
+        self._discourse_url = discourse_url
+        self._api_key = api_key
+        self._api_username = api_username
+
+    @ndb.tasklet
+    def _getRequest(self, req_string, params=None, payload=None):
+        response = yield self._sendDiscourseRequest(req_string, params,
+                                                    payload, 'GET')
```

Comment:
Sure, will do! Thanks for the feedback, btw -- this is still very much a work in progress, so any advice is appreciated. I'm working next on refactoring the different areas (users, categories, groups) into different submodules.